netzob/src/netzob/Inference/Vocabulary/FormatOperations/FieldOperations.py

```diff
@@ -147,11 +147,95 @@ def mergeFields(self, field1, field2):
                 "Field1 must be directly on the left of field2 (iField1={0}, iField2={1})".
                 format(iField1, iField2))
 
-        # build a new field domain
-        newDomain = Agg([field1.domain, field2.domain])
-        newField = Field(domain=newDomain, name="Merge")
-        newField.encodingFunctions = list(field1.encodingFunctions.values())
+        try:
+            # first try to completely blend the fields if this is supported
+            newField = FieldOperations._blendFields(field1,field2)
+        except TypeError or NotImplementedError:
```

Comment:
The [`except` clause](https://docs.python.org/3/reference/compound_stmts.html#except) does not support the `… or …` expression to handle a list of exceptions at the same level, as the `or` clause would be solved to an atomic expression (`TypeError` in this case, silently bypassing the other) at typing time.

I guess you wanted to write this instead, to catch both exceptions:
```python
except (TypeError, NotImplementedError):
```

Reply:
Thank you very much for the hint. By looking at the issue more closely, I noticed another of my errors in exception handling. The next commit should fix both.